diff --git a/src/system_error.cpp b/src/system_error.cpp
index 87f35ae..1e54f01 100644
--- a/src/system_error.cpp
+++ b/src/system_error.cpp
@@ -63,7 +63,7 @@ constexpr size_t strerror_buff_size = 1024;
 string do_strerror_r(int ev);
 
 #if defined(__linux__) && !defined(_LIBCPP_HAS_MUSL_LIBC)                      \
-    && (!defined(__ANDROID__) || __ANDROID_API__ >= 23)
+    && (!defined(__ANDROID__) || __ANDROID_API__ >= 23) || defined(_NEWLIB_VERSION)
 // GNU Extended version
 string do_strerror_r(int ev) {
     char buffer[strerror_buff_size];
